update-snap/action.yml

```diff
@@ -0,0 +1,136 @@
+name: 'Update snap tag'
+description: 'Update the snapcraft.yaml for a given snap with the latest upstream tag'
+
+# Inputs required by the action to perform correctly.
+# token: This is the GITHUB_TOKEN that exists for the other repo whose workflow uses this action. Without the other repo's token, this action does not have permission to perform tasks on the other repo, like pushing changes or opening a PR
+# repo: This is the repo who is using this action. This repo contains the snapcraft.yaml that this action will be working on.
+inputs:
+  token:
+    description: 'A Github PAT'
+    required: true
+  repo:
+    description: 'The repo containing the snapcraft.yaml to be updated'
+    required: true
+    default: 'None'
+  version-schema:
+    description: 'Version schema of snapping repository'
+    required: false
+  rock-version-schema:
+    description: 'Version schema of rock repository'
+    required: false
+  yaml-path:
+    description: 'Path to the YAML file'
+    required: false
+  readme-path:
+    description: 'Path to the README file'
+    required: false
+
+# Outputs generated by running this action.
+outputs:
+  new-snapcraft-yaml:
+    description: "Updated snapcraft.yaml"
+    value: ${{ steps.updatesnapyaml.outputs.new-snapcraft-yaml }}
+
+# Global env var that can keep track of if there was a change or not. This determines if there are commits to be pushed.
+env:
+  IS_CHANGE: false
+  IS_VERSION_CHANGE: false
+  IS_README_CHANGE: false
+
+# The jobs description defining a composite action
+runs:
+  using: "composite"
+
+  steps:
+# Step to checkout the desktop-snaps repo so that we can run updatesnapyaml.py. We need to define a folder called desktop-snaps to avoid the checkout overwriting the other repo's checkout which comes first in the other repo's workflow
+    - name: checkout desktop-snaps repo
+      uses: actions/checkout@v4
+      with:
+        repository: 'ubuntu/desktop-snaps'
+        ref: stable
+        path: desktop-snaps
+
+# Step to run the script that will generate a new output_file with an updated tag, if one is available. If there was a change, then we move this to the snapcraft.yaml and note that there was a change. 
+    - name: run updatesnapyaml
+      id: updatesnapyaml
+      run: |
+        set -x
+        ./desktop-snaps/updatesnap/updatesnapyaml.py \
```

Comment:
the script `updatesnapyaml.py` does the actual dirty work of discovering upstream versions and comparing them with the currently used ones. It also needs to get transferred into `snapcrafters/ci`.